### PR TITLE
Switch distribution to new style Requests

### PR DESCRIPTION
**Pull request by nik9000**
In #29623 we added `Request` object flavored requests to the low level
REST client and in #30315 we deprecated the old `performRequest`s. This
changes all calls in the `distribution/archives/integ-test-zip` project
to use the new versions.

### CODE REVIEW

File: distribution/archives/integ-test-zip/src/test/java/org/elasticsearch/test/rest/NodeRestUsageIT.java [hunk 1]
```diff
@@ -80,24 +81,28 @@ public void testWithRestUsage() throws IOException {
...
-        client().performRequest("GET", "/_nodes");
-        client().performRequest("GET", "/test/_search");
-        client().performRequest("GET", "/_nodes/stats");
-        client().performRequest("DELETE", "/test");
+        client().performRequest(new Request("PUT", "/test"));
+        Request request = new Request("POST", "/test/doc/1");
+        request.setJsonEntity("{\"foo\": \"bar\"}");
+        client().performRequest(request);
+        request = new Request("POST", "/test/doc/2");
+        request.setJsonEntity("{\"foo\": \"bar\"}");
```
Comment (jasontedor) [inline distribution/archives/integ-test-zip/src/test/java/org/elasticsearch/test/rest/NodeRestUsageIT.java]: I really dislike this style of variable reuse in our tests. If I use my IDE to navigate to the definition of this variable I end up on a line assigning a value to this variable that is removed from its current value. This hinders readability, especially in longer tests. Let’s avoid introducing it here, we should be moving away from it.

**Review by jasontedor — CHANGES_REQUESTED** [review]
I left a general thought.

File: distribution/archives/integ-test-zip/src/test/java/org/elasticsearch/test/rest/NodeRestUsageIT.java (hunk 1 above)
Comment (nik9000) [reply distribution/archives/integ-test-zip/src/test/java/org/elasticsearch/test/rest/NodeRestUsageIT.java]: Do you prefer extra blocks or different names to stop the reuse?

**Comment by nik9000** [comment]
@jasontedor, could you have another look when you get a chance?

**Comment by nik9000** [comment]
@jasontedor , could you have another look at this when you get a chance?

**Review by jasontedor — APPROVED** [review]
Yeah, that looks good. Thanks.

**Pull request merged**